### PR TITLE
`Docs` Card.tsx

### LINKED ISSUE #10150: Create a <Card> component that can be used in a `.md` file
Create a primitive JSX Card component to be used across the site and for markdown content like in this figma example [`Voice and Tone` ](https://www.figma.com/file/vO0wToyBpYjeIwr8doAAWT/Uplift-polaris.shopify.com?type=design&node-id=37%3A25215&mode=design&t=Z6KugRKI8PLcmyHr-1)

If it were possible it'd be ideal to use the polaris-react `Card`, but we're not able to support dark mode for that card. We should build a similar lightweight card component that's able to wrap any sort of children

Image

#### [Background card](https://www.figma.com/file/vO0wToyBpYjeIwr8doAAWT/Uplift-polaris.shopify.com?type=design&node-id=50%3A24886&mode=design&t=kQj4os2WflWtJs5B-1)

Image

#### [Media card (vertical and horizontal)](https://www.figma.com/file/vO0wToyBpYjeIwr8doAAWT/Uplift-polaris.shopify.com?type=design&node-id=50%3A4050&mode=design&t=kQj4os2WflWtJs5B-1)

Image

#### [Do / Dont card](https://www.figma.com/file/vO0wToyBpYjeIwr8doAAWT/Uplift-polaris.shopify.com?type=design&node-id=53%3A27059&mode=design&t=kQj4os2WflWtJs5B-1)

Image

```

### PR DESCRIPTION
**Pull request by gwyneplaine**
### WHY are these changes introduced?

Fixes #10150  

### WHAT is this pull request doing?

Create a light-weight card component with the following: 
* responsive prop for `padding` 
* optional `className` prop for overrides. 

### Things I've left out: 
* `roundedAbove` breakpoints are subtly different in polaris-react vs polaris.shopify.com, so porting this functionality over is not a straightforward exercise. I think the right call is to get the MVP lightweight card in so we can start using and iterating. We can add breakpoint parity in in a separate PR. 
* `backgroundColor`, not sure we need this one given the screenshots and designs in the linked issue, but happy to add this in later if I'm mistaken 👍 

### Things that are subtly different: 
* In polaris-react `padding` is just a mirror of the responsiveProp functionality in`Box`, in this PR I've chosen to have that functionality localized to `Card` for now, to avoid having to worry about accidentally breaking `polaris.shopify.com` Box usage in other areas of the docs. (not a strongly held opinion though)

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by jesstelford** [comment]
Heads up that the `responsive-props` SASS mixins are undergoing some enhancements in https://github.com/Shopify/polaris/pull/10404/files#diff-eca995eb337e5fc8efe32c386c6a1d75efe5ee8cf67bd5bc0fbd6ba9d119d2fd & https://github.com/Shopify/polaris/pull/10489/files#diff-eca995eb337e5fc8efe32c386c6a1d75efe5ee8cf67bd5bc0fbd6ba9d119d2fd so these might be out of date soon.

**Comment by gwyneplaine** [comment]
> Heads up that the `responsive-props` SASS mixins are undergoing some enhancements in https://github.com/Shopify/polaris/pull/10404/files#diff-eca995eb337e5fc8efe32c386c6a1d75efe5ee8cf67bd5bc0fbd6ba9d119d2fd & https://github.com/Shopify/polaris/pull/10489/files#diff-eca995eb337e5fc8efe32c386c6a1d75efe5ee8cf67bd5bc0fbd6ba9d119d2fd so these might be out of date soon.

Happy to make this adjustment once that lands in `next` 👍

**Comment by gwyneplaine** [comment]
Example of usage in a `.md` document. 
Screenshot 2023-09-18 at 4 25 52 pm

File: polaris.shopify.com/src/components/Card/Card.module.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  // As this will be used with Markdown elements and components
+  // We want to preserve margin rules for all children
+  // Except the last-child, so as not to introduce unecessary space
+  // at the bottom of the card
+  > :last-child {
+    margin-bottom: 0;
+  }
+
+  box-shadow: 0px 1px 0px 0px rgba(26, 26, 26, 0.07),
```
Comment (gwyneplaine) [inline polaris.shopify.com/src/components/Card/Card.module.scss:29]: these values are taken straight from Figma, I'm unclear if these should replace the existing [--card-shadow token](https://github.com/Shopify/polaris/blob/040740989250a57889a16a22f599fb7452cfe68a/polaris.shopify.com/src/styles/globals.scss#L208), I've left it localised for now.

File: polaris.shopify.com/src/styles/mixins.scss [hunk 2]
```diff
@@ -28,3 +30,61 @@
...
+  // stylelint-disable -- Polaris component custom properties
+  --pc-#{$componentName}-#{$componentProp}-xs: initial;
+  --pc-#{$componentName}-#{$componentProp}-sm: initial;
+  --pc-#{$componentName}-#{$componentProp}-md: initial;
+  --pc-#{$componentName}-#{$componentProp}-lg: initial;
+  --pc-#{$componentName}-#{$componentProp}-xl: initial;
+
+  #{$declarationProp}: var(--pc-#{$componentName}-#{$componentProp}-xs);
+
+  @media (min-width: $breakpointMobile) {
```
Comment (gwyneplaine) [inline polaris.shopify.com/src/styles/mixins.scss:44]: breakpoint taxonomy and values are slightly different in the docs site compared to polaris-react
Let me know if these should be unified going forward, I can look at that work in a separate PR.

File: polaris.shopify.com/src/components/Card/Card.module.scss (hunk 1 above)
Comment (yurm04) [reply polaris.shopify.com/src/components/Card/Card.module.scss:29]: i think this is fine for now, we can update this value once we've finalized token value updates

File: polaris.shopify.com/src/styles/mixins.scss (hunk 2 above)
Comment (yurm04) [reply polaris.shopify.com/src/styles/mixins.scss:44]: I think this is good imo not critical to map to polaris-react

**Review by yurm04 — APPROVED** [review]
Looks good ❇️ thank you, Charles!!

**Pull request merged**